server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -540,10 +543,9 @@ private static void parseArray(ParseContext context, ObjectMapper parentMapper,
             Tuple<Integer, ObjectMapper> parentMapperTuple = getDynamicParentMapper(context, paths, parentMapper);
             parentMapper = parentMapperTuple.v2();
             ObjectMapper.Dynamic dynamic = dynamicOrDefault(parentMapper, context);
```

Comment:
`dynamicOrDefault` has this:
```
            if (parentMapper == null) {
                // If parentMapper is ever null, it means the parent of the current mapper was dynamically created.
                // But in order to be created dynamically, the dynamic setting of that parent was necessarily true
                return ObjectMapper.Dynamic.TRUE;
            }
```
which worries me a bit. Should that depend on the configured dynamic value?

Reply:
It might not be a problem, but it is worth an extra comment explaining why not.

Reply:
well spotted, I actually need to double check if this may cause troubles, I am not 100% sure yet.

Reply:
this was indeed a bug, should be fixed now but I had to add a way to retrieve dynamically created objects, as we can no longer assume TRUE now, it could also be RUNTIME.